### ISSUE TITLE
[BUG] Массовые запросы /history

### ISSUE THREAD

**Issue by Extred** [alor-broker/Astras-Trading-UI#1534]
### Опишите баг

Если открыть график по малоликвидной бумаге (пример - RU000A1057A0), то график долго грузится и не отзывается пока грузится.
В консоли видны сотни запросов /history, в т.ч. с некорректными таймстемпами.

### Воспроизведение

- Окрыть виджет графика TV
- Выбрать бумагу RU000A1057A0
- Посмотреть dev tools - network

### Среда.

- win11
- Chrome 121

### Ожидаемое поведение

График не зависает. Отправляется один или несколько запросов (не сотни)

### Скриншоты

<img width="914" alt="image" src="https://github.com/alor-broker/Astras-Trading-UI/assets/3522038/322aca39-0cf8-4f18-bcb2-f1885f45a900">

### Правила

- [X] Я согласен следовать правилам

**Comment by sbelashevskiy** [comment]
Не вижу ни локально ни на скриншоте каких-либо "некооректных" таймстепов.

TV график на данном крайнем случае работает точно также, как https://tradingview.alorbroker.ru/

Мы не контролируем какие границы запрашивает график, это делает сторонняя либа по своему внутреннему алгоритму, который учитывает масштаб элементов, протяженность оси времени и текущий таймфрэйм:
https://www.tradingview.com/charting-library-docs/latest/connecting_data/Datafeed-API#getbars
![image](https://github.com/alor-broker/Astras-Trading-UI/assets/29191749/cf17cb3d-4920-4811-afba-b429d14a09e2)

Либа сама вычисляет границы данных, которые ей нужны для заполнения пространства графика.
Алгоритм либы явно учитывает факт, что **на разных фреймах объем данных может быть разным**.
Мы не можем просто так взять и сократить число запросов просто путем расширения запрашиваемых границ - на одной границе данных может не быть, а в другую попадет 1M+ точек и все это будет передаваться по сети.

API позволяет передавать ТОЛЬКО границы интервала, а корректного решения данной проблемы еще и нужно обрабатывать мин. кол-во точек, которое необходимо либе
![image](https://github.com/alor-broker/Astras-Trading-UI/assets/29191749/a44c0d31-7e6c-4532-87e2-4947446523dd).

По сути на таком крайнем случае либа генерирует запросы, которые сканируют ось времени в поисках данных.
Например, на SBER такой необходимости у либы не возникает - имеющихся данных хватает, чтобы заполнить ось  времени.

Данная проблема должна решаться путем расширения API - добавлением доп. параметров.
На фронте она может быть решена только путем сканирования данных, что сейчас и делается.
На фронте мы понятия не имеем какие данные есть, где будет провал, и где при расширении границ мы будем получать огромные массивы данных, которые либо невозможно будет передать по сети, либо график будет обрабатывать их еще дольше

**Comment by Extred** [comment]
Судя по всему параметр "countBack" как раз для решения этой проблемы TV и ввели. В старых версиях либы использовались "next" и "prev", но сейчас в доке они даже не упоманиются.

Вы можете прокинуть countBack в запросы с фронта? Его же либа TV генерит?
А коллег уже на бэке допилят новый параметр. Полагаю, это заодно и другую проблему решит - когда после выходных на мелких таймфреймах пустой график в TV.

**Comment by sbelashevskiy** [comment]
Да, можем прокинуть. Но, думаю, лучше сначала на бэке сделать  потом уже на фронте добавить, чтоб сразу проверить.
Добавлю тогда на эту задачу label Backend, чтобы задача создалась

**Comment by sbelashevskiy** [comment]
Список инструментов тоже запрашивает историю и тоже "мучается" с поиском последней записи
